raiden/network/transport/matrix/transport.py

```diff
@@ -1064,81 +964,20 @@ def _handle_invite(self, room_id: RoomID, state: dict) -> None:
-                room_id=room_id,
-                user=user,
-            )
+            # Invalid invite, ignore
             return
 
-        sender_join_events = [
-            event
-            for event in state["events"]
-            if event["type"] == "m.room.member"
-            and event["content"].get("membership") == "join"
-            and event["state_key"] == sender
-        ]
-
-        if not sender_join_events or not sender_join_events[0]:
-            self.log.debug("Invite: no sender join event", room_id=room_id)
-            return  # there should always be one and only one join membership event for the sender
+        sender = invite_events[0]["sender"]
 
-        join_rules_events = [
-            event for event in state["events"] if event["type"] == "m.room.join_rules"
-        ]
+        self.log.debug("Rejecting invite", room_id=room_id, sender=sender)
 
-        # room privacy as seen from the event
-        private_room: bool = False
-        if join_rules_events:
-            join_rules_event = join_rules_events[0]
-            private_room = join_rules_event["content"].get("join_rule") == "invite"
-
-        # Ignore the room if it is not private, since that can be an attack
-        # vector, e.g. secret reveal messages would be available to any user that
-        # knows the room id. (only private rooms are used since ce246af806)
-        # this also filters invites to broadcast rooms
-        if private_room is False:
-            self.log.debug("Invite: ignoring room since it is not private", room_id=room_id)
-            return
-
-        room = None
-        # try to join the room
-        try:
-            room = self.retry_api_call(self._client.join_room, room_id_or_alias=room_id)
-        except MatrixRequestError as ex:
-            # this is catching invitation to room of invalid server -> rejecting the invite
-            if ex.code == 404 and ex.content == {
-                "errcode": "M_UNKNOWN",
-                "error": "No known servers",
-            }:
-                # reject invite by "leaving" the room
-                dummy_room = Room(self._client, room_id)
-                self._leave_unexpected_rooms([dummy_room])
-
-        assert room is not None, f"joining room {room} failed"
-
-        self.log.debug(
-            "Joined from invite",
+        self.retry_api_call(
+            self._client.api.send_state_event,
             room_id=room_id,
-            canonical_alias=room.canonical_alias,
-            inviting_address=to_checksum_address(peer_address),
+            event_type="m.room.member",
+            content={"membership": "leave"},
         )
 
-        # room state may not populated yet, so we populate 'invite_only' from event
-        room.invite_only = private_room
-        self._set_room_id_for_address(address=peer_address, room_id=room_id)
-
     def _handle_member_join(self, room: Room) -> None:
```

Comment:
Do we need the handle member join logic at all?

Reply:
Hm, yeah I guess that falls into the same category as the remaining room logic in the message handling.
It does somewhat protect against accidental changes, but I have nothing against removing it either.

Reply:
AFAIR, it only checked for forbidden members in a room. This should already be unreachable code, since we reject invites and leave all rooms in the beginning